Doc/library/typing.rst

```diff
@@ -623,14 +623,21 @@ These can be used as types in annotations using ``[]``, each having a unique syn
 
 .. data:: Union
 
-   Union type; ``Union[X, Y]`` is equivalent to ``X | Y`` and means either X or Y.
+   Union type is defined as ``X | Y`` and means either X or Y.
 
-   To define a union, use e.g. ``Union[int, str]`` or the shorthand ``int | str``.  Details:
+   There's also an older version ``Union[int, str]``.
+   Note that the ``X | Y`` is preferred in readability in most cases.
+   However, there are :ref:`specified runtime differences<types-union>`.
+
+   Details:
 
    * The arguments must be types and there must be at least one.
```

Comment:
arguments are only present when using the Union[X, Y] syntax.  Prepend a clarification on this "When using Union[] syntax, the arguments ..."

Add something that covers what may be |ed together.  "Only types may be used with the | operator for this purpose" or similar perhaps.

Reply:
This text is technically in the "module contents" section. There is more on `X|Y` in the docs for the `types` module. Maybe we could link there (on the word "shorthand") like was done in the first version of this PR. If the description of `X|Y` there is not clear enough it should be updated there IMO.

Reply:
Crosslinking shorthand does make sense.  Otherwise I think the _text_ here is clear enough as is.